### PR TITLE
INTL-115: Make the intl_message_migration also look for likely user-visible String constants

### PR DESCRIPTION
**Pull request by alanknight-wk**
## Motivation
  
There are lots of 
   String const cancel => 'Cancel';
that are likely user-visible. Look for these, as const strings that start with an upper case level. Exclude the common case of date formats, which may start with doubled upper-case letters, e.g. 'MM/dd/YYYY'

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/executables/intl_message_migration.dart [hunk 1]
```diff
@@ -78,6 +79,13 @@ void main(List<String> args) async {
       negatable: false,
       help: 'Forces ansi color highlighting of stderr. Useful for debugging.',
     )
+    ..addFlag(
+      _migrateConstants,
+      negatable: true,
+      defaultsTo: false,
```
Comment (robbecker-wf) [inline lib/src/executables/intl_message_migration.dart]: I must ask .. should we make this default to true?

File: lib/src/executables/intl_message_migration.dart [hunk 2]
```diff
@@ -221,11 +229,14 @@ void main(List<String> args) async {
...
     final intlPropMigrator = IntlMigrator(className, outputFile);
     final displayNameMigrator = ConfigsMigrator(className, outputFile);
+    final constantStringMigrator =
+        ConstantStringMigrator(className, outputFile);
     final importMigrator =
         (FileContext context) => intlImporter(context, packageName, className);
 
     exitCode = await runCodemodSequences(packageDartPath, [
       [intlPropMigrator],
+      if (parsedArgs[_migrateConstants]) [constantStringMigrator],
```
Comment (robbecker-wf) [inline lib/src/executables/intl_message_migration.dart]: nice use of conditionals in a list 👍

File: test/executables/intl_message_migration_test.dart [hunk 3]
```diff
@@ -24,6 +24,11 @@ import 'mui_migration_test.dart' show testCodemod;
 // the stdout/stderr of the codemod processes.
 final _debug = false;
 
+// The help text may have different amount of whitespace depending on the names
+// of the options, so collapse all whitespace to a
```
Comment (robbecker-wf) [inline test/executables/intl_message_migration_test.dart:28]: this comment seems incomplete?

File: test/intl_suggestors/constant_migrator_test.dart [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+import '../resolved_file_context.dart';
+import '../util.dart';
+
+void main() {
+  final resolvedContext = SharedAnalysisContext.overReact;
+
+  // Warm up analysis in a setUpAll so that if getting the resolved AST times out
+  // (which is more common for the WSD context), it fails here instead of failing the first test.
+  setUpAll(resolvedContext.warmUpAnalysis);
```
Comment (robbecker-wf) [inline test/intl_suggestors/constant_migrator_test.dart:14]: "Today I learned"
This is a neat trick, where did you learn this?

File: test/intl_suggestors/constant_migrator_test.dart [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+          expectedOutput: '''
+            final String foo = TestClassIntl.iAmAUservisibleConstant;
+            ''',
+        );
+        final expectedFileContent =
+            '\n  static String get iAmAUservisibleConstant => Intl.message(\'I am a user-visible constant\', name: \'TestClassIntl_iAmAUservisibleConstant\',);';
+        expect(file.readAsStringSync(), expectedFileContent);
+      });
+
+      test('non-matching standlalone constant', () async {
```
Comment (robbecker-wf) [inline test/intl_suggestors/constant_migrator_test.dart:49]: speling :)
```suggestion
      test('non-matching standalone constant', () async {
```

File: lib/src/executables/intl_message_migration.dart (hunk 1 above)
Comment (alanknight-wk) [reply lib/src/executables/intl_message_migration.dart]: I was going to hide it behind a flag until we had that discussion.

File: test/intl_suggestors/constant_migrator_test.dart (hunk 4 above)
Comment (alanknight-wk) [reply test/intl_suggestors/constant_migrator_test.dart:14]: From the mui_migration_test.dart file that I copied it from. So probably Greg or Aaron?

File: lib/src/executables/intl_message_migration.dart (hunk 1 above)
Comment (robbecker-wf) [reply lib/src/executables/intl_message_migration.dart]: lovely. let's leave it behind a flag for now but I think we'll probably want to move it to default if the results we get from this addition are generally good.

**Review by robbecker-wf — APPROVED** [review]

**Comment by robbecker-wf** [comment]
QA+1 CI Passes and verified it replaces constants with the flag enabled and vice versa.

**Comment by robbecker-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**